src/Container/ApplicationFactory.php

```diff
@@ -165,9 +166,20 @@ private function injectRoutes(Application $app, ContainerInterface $container)
             $methods = (isset($spec['allowed_methods']) && is_array($spec['allowed_methods']))
                 ? $spec['allowed_methods']
                 : null;
-            $name    = isset($spec['name']) ? $spec['name'] : null;
+            $name = isset($spec['name']) ? $spec['name'] : null;
             $methods = (null === $methods) ? Route::HTTP_METHOD_ANY : $methods;
-            $route   = new Route($spec['path'], $spec['middleware'], $methods, $name);
+
+            if (is_array($spec['middleware'])) {
```

Comment:
That prevents people from specifiyng `[SomeClass::class, 'someMethod']` as callable middleware.